File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetPhysicalDevicePresentRectanglesKHX.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetPhysicalDevicePresentRectanglesKHX_call(JNIEnv* env, jclass jc, jlong address, jlong physicalDevice, jlong surface, jlong pRectCount, jlong pRects, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkGetPhysicalDevicePresentRectanglesKHX f = (PFN_vkGetPhysicalDevicePresentRectanglesKHX)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkPhysicalDevice*)jniLongToPointer(physicalDevice)),
        *((VkSurfaceKHR*)jniLongToPointer(surface)),
        ((uint32_t*)jniLongToPointer(pRectCount)),
        ((VkRect2D*)jniLongToPointer(pRects))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkGetPhysicalDevicePresentRectanglesKHX.java
package cz.mg.vulkan;

public class PFNvkGetPhysicalDevicePresentRectanglesKHX extends VkFunctionPointer {
    public PFNvkGetPhysicalDevicePresentRectanglesKHX() {
    }

    protected PFNvkGetPhysicalDevicePresentRectanglesKHX(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetPhysicalDevicePresentRectanglesKHX(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetPhysicalDevicePresentRectanglesKHX(long value) {
        setValue(value);
    }

    public PFNvkGetPhysicalDevicePresentRectanglesKHX(VkInstance instance) {
        super(instance, new VkString("vkGetPhysicalDevicePresentRectanglesKHX"));
    }

    public void call(VkPhysicalDevice physicalDevice, VkSurfaceKHR surface, VkUInt32 pRectCount, VkRect2D pRects, VkResult rval){
        call(getValue(), physicalDevice != null ? physicalDevice.getVkAddress() : VkPointer.getNullAddressNative(), surface != null ? surface.getVkAddress() : VkPointer.getNullAddressNative(), pRectCount != null ? pRectCount.getVkAddress() : VkPointer.NULL, pRects != null ? pRects.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void call(long vkaddress, long physicalDevice, long surface, long pRectCount, long pRects, long rval);
}
